docs/index.md

```diff
@@ -7,7 +7,29 @@ has_children: true
 
 # pint
 
-pint is a Prometheus rule linter.
+pint is a Prometheus rule linter/validator.
+
+## Requirements
+
+pint will run checks on Prometheus alerting & recording rules to detect potential problems
+with those rules.
+Some checks rely only on the rule itself and can be run "offline" - without talking to any
+Prometheus server.
+You can run pint in "offline" if you:
+- Don't pass any configuration file to pint.
+- You pass configuration file to pint that **doesn't** contain any `prometheus` definition.
+- You pass `--offline` flag to `pint` command.
+
+Most checks included in pint will require sending queries to a running Prometheus server where
+those rules are, or would be, deployed.
+Those checks are enabled if you pass a configuration file to pint that includes at least one
+`prometheus` block will.
+Checks might use various Prometheus
+[HTTP API endpoints](https://prometheus.io/docs/prometheus/latest/querying/api/) to retrive
+extra information, for example Prometheus configuration or metrics metadata.
+If you run pint against a different service, like [Thanos](https://thanos.io/) some checks
+might return an erorr, since not all Prometheus HTTP APIs are supported by it.
+In that case you might want to disable failing checks in pint configuration file.
```

Comment:
Maybe add something like "failing checks will be reported as `pint_problem`" to point users in the right direction

Reply:
fix in #576